lib/jxl/enc_fast_lossless.h

```diff
@@ -18,10 +17,32 @@
 #endif
 #endif
 
+#if !FJXL_STANDALONE
+#include <jxl/encode.h>
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#if FJXL_STANDALONE
+// Simplified version of the streaming input source from jxl/encode.h
+// We only need this part to wrap the full image buffer in the standalone mode
+// and this way we don't need to depend on the jxl headers.
+struct JxlChunkedFrameInputSource {
+  void* opaque;
+  const void* (*get_color_channel_data_at)(void* opaque, size_t xpos,
+                                           size_t ypos, size_t xsize,
+                                           size_t ysize, size_t* row_offset);
+  void (*release_buffer)(void* opaque, const void* buf);
+};
+// The standalone version does not use this struct, but we define it here so
+// that we don't have to clutter all the function signatures with defines.
+struct JxlEncoderOutputProcessorWrapper {
+  int palceholder;
```

Comment:
typo: placeholder

I assume this is not an empty struct b/c some ABI difference between C and C++? I'd probably rename it to `unused_` if so

Reply:
Yes, because of the ABI difference. I renamed it to unused.